lib/modules/manager/puppet/extract.ts

```diff
@@ -0,0 +1,63 @@
+import { logger } from '../../../logger';
+import { PuppetDatasource } from '../../datasource/puppet';
+import type { PackageDependency, PackageFile } from '../types';
+import { forgeRegexFactory, simpleModuleLineRegexFactory } from './constants';
+
+interface ForgeContent {
+  forgeUrl: string;
+  moduleContent: string;
+}
+
+export function extractPackageFile(content: string): PackageFile | null {
+  logger.trace('puppet.extractPackageFile()');
+
+  const deps: PackageDependency[] = [];
+
+  const forgeContents: ForgeContent[] = [];
+
+  const forgeRegex = forgeRegexFactory();
+  let forge: RegExpExecArray;
+
+  if (!forgeRegexFactory().test(content)) {
+    forgeContents.push({
+      forgeUrl: 'https://forgeapi.puppet.com', // TODO: default Registry
```

Comment:
Should be defined on datasource as defaultUrl.

Reply:
done